main/pipeline/pipeline.cc

```diff
@@ -277,30 +274,33 @@ unique_ptr<parser::Node> runRBSRewrite(core::GlobalState &gs, core::FileRef file
     return node;
 }
 
-unique_ptr<parser::Node> runPrismParser(core::GlobalState &gs, core::FileRef file, const options::Printers &print) {
+parser::Parser::ParseResult runPrismParser(core::GlobalState &gs, core::FileRef file, const options::Printers &print) {
     Timer timeit(gs.tracer(), "runParser", {{"file", string(file.data(gs).path())}});
 
-    unique_ptr<parser::Node> nodes;
-    {
+    parser::Parser::ParseResult parseResult = [&]() {
         core::MutableContext ctx(gs, core::Symbols::root(), file);
         core::UnfreezeNameTable nameTableAccess(gs); // enters strings from source code as names
-        nodes = parser::Prism::Parser::run(ctx);
-    }
+        if (gs.cacheSensitiveOptions.rbsEnabled) {
+            return parser::Prism::Parser::run(ctx, false);
+        } else {
+            return parser::Prism::Parser::run(ctx);
+        }
```

Comment:
Could we just always pass the bool?

Also, it's worth commenting to explain why we're turning it off

```suggestion
        // The RBS rewriter produces plain Whitequark nodes and never `NodeWithExpr`,
        // even ones that we no longer produce in direct-translation mode.
        // For now, disable all direct translation, and fallback to `Desugar.cc`
        auto directlyTranslate = !gs.cacheSensitiveOptions.rbsEnabled
        return parser::Prism::Parser::run(ctx, directlyTranslate);
```